DigitalLearningSolutions.Data.Tests/Services/GroupServiceTests/GroupsServiceAddCourseTests.cs

```diff
@@ -0,0 +1,597 @@
+        {
+            // Given
+            const int adminId = 1;
+            var groupCourse = GroupTestHelper.GetDefaultGroupCourse(supervisorAdminId: null);
+            SetupEnrolProcessFakes(
+                GenericNewProgressId,
+                GenericRelatedTutorialId
+            );
+            SetUpAddCourseEnrolProcessFakes(groupCourse);
+
+            // When
+            groupsService.AddCourseToGroup(
+                reusableGroupCourse.GroupId,
+                reusableGroupCourse.CustomisationId,
+                8,
+                adminId,
+                true,
+                null
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                DelegateProgressRecordMustNotHaveBeenUpdated();
+                A.CallTo(
+                    () => progressDataService.CreateNewDelegateProgress(
+                        reusableGroupDelegate.DelegateId,
+                        reusableGroupCourse.CustomisationId,
+                        reusableGroupCourse.CurrentVersion,
+                        testDate,
+                        3,
+                        adminId,
+                        A<DateTime?>._,
+                        0
+                    )
+                ).MustHaveHappened();
+                A.CallTo(() => progressDataService.CreateNewAspProgress(GenericRelatedTutorialId, GenericNewProgressId))
+                    .MustHaveHappened();
+            }
+        }
+
+        [Test]
+        public void
+            AddCourseToGroup_add_new_progress_record_uses_course_supervisor_id_if_course_supervisor_is_not_null()
+        {
+            // Given
+            const int adminId = 1;
+            const int supervisorId = 14;
+            var groupCourse = GroupTestHelper.GetDefaultGroupCourse(supervisorAdminId: supervisorId);
+            SetupEnrolProcessFakes(
+                GenericNewProgressId,
+                GenericRelatedTutorialId
+            );
+            SetUpAddCourseEnrolProcessFakes(groupCourse);
+
+            // When
+            groupsService.AddCourseToGroup(
+                reusableGroupCourse.GroupId,
+                reusableGroupCourse.CustomisationId,
+                8,
+                adminId,
+                true,
+                supervisorId
+            );
+
+            // Then
+            using (new AssertionScope())
+            {
+                DelegateProgressRecordMustNotHaveBeenUpdated();
+                A.CallTo(
+                    () => progressDataService.CreateNewDelegateProgress(
+                        reusableGroupDelegate.DelegateId,
+                        reusableGroupCourse.CustomisationId,
+                        reusableGroupCourse.CurrentVersion,
+                        testDate,
+                        3,
+                        adminId,
+                        testDate.AddMonths(12),
+                        supervisorId

```

Comment:
Similar to a comment below in the file, I think the supervisor Id and test set up might need double checking. I don't think it's obvious what is supposed to be happened in this test.